BackendBench/eval.py

```diff
@@ -64,6 +66,21 @@ def allclose(a, b, atol=1e-2, rtol=1e-2):
         return False
 
 
+def equal_metadata(a, b):
+    try:
+        _allclose(a.shape, b.shape, atol=0.0, rtol=0.0)
+        _allclose(a.stride(), b.stride(), atol=0.0, rtol=0.0)
+        _allclose(a.dtype, b.dtype, atol=0.0, rtol=0.0)
+        _allclose(a.device, b.device, atol=0.0, rtol=0.0)
+        return True
```

Comment:
I'd check the type string as well as per             self.assertEqualTypeString(torch.empty_like(a), a)